pytradfri/device/air_purifier.py

```diff
@@ -35,25 +36,25 @@ def raw(self) -> TypeRaw:
         )
 
     @property
-    def mode(self) -> int:
-        """Return the current mode of the air purifier.
-
-        0: off
-        1: Fan level auto
-        10: Fan level 1
-        20: Fan level 2
-        30: Fan level 3
-        40: Fan level 4
-        50: Fan level 5
+    def is_auto_mode(self) -> bool:
         """
-        return cast(int, self.raw[ATTR_AIR_PURIFIER_MODE])
+        Return auto mode on or off.
+
+        Auto mode sets the fan speed automatically based on the air quality.
+        """
+        return self.raw[ATTR_AIR_PURIFIER_MODE] == ATTR_AIR_PURIFIER_MODE_AUTO
+
+    @property
+    def state(self) -> bool:
+        """Return device state, ie on or off."""
+        return cast(int, self.raw[ATTR_AIR_PURIFIER_MODE]) > 0
```

Comment:
Is it never interesting to know the mode above 1?

Reply:
Please see the screen shot from the official app below. The way I see it, there is one mode (auto). Alternatively, the user sets the fan speed using the slider.

Reply:
Ok, but we're now hiding this api info, making it possible to access it, right?

Reply:
We expose auto mode, we expose on/off and we expose the fan speed. Shouldn’t that cover all use cases? Using the previous “mode” only set the fan speed to some fixed values.

Reply:
Thanks for developing this. Talking about exposing, would exposing air quality be possible? If an Fr for that is needed I d be glad to write that.

Reply:
I'm asking you if you think this. Is your answer that it never will be interesting to know the mode?

@Mariusthvdb please don't divert the conversation to another topic. If you want to ask a question, do that in a separate conversation.

Reply:
No, in the way the device currently work I find the suggested changes to be the most appropriate. I cannot see why we should to consider the fan speed as being modes.

Reply:
Ok.